transport/pool.go

```diff
@@ -118,6 +129,13 @@ func (r *PoolRefiller) initConnPool() error {
 	}
 	ss := s.ScyllaSupported()
 
+	if s, ok := s.Options["SCYLLA_SHARD_AWARE_PORT"]; ok && !strings.HasSuffix(r.addr, ":"+s[0]) {
+		err = changePort(&r.addr, &s[0])
+		if err != nil {
+			return fmt.Errorf("port change: %w", err)
+		}
+	}
+
```

Comment:
This is very puzzling change.

I see you try to start with Shard Aware port, it's not needed IMO.

Reply:
So can we assume (for now) that we are given shard aware ports while creating session?
For example, TestNodeConnPoolIntegration won't pass if we create pool with port 9042 instead of 19042
(we won't always be able to create connections to every shard since it does not depend on us in this scenario).

In general it is quiet easy to get shard aware port after connecting to a node (in contrary to getting any port from a node that we are not connected to yet) so I would consider updating shard aware port.